source/plugin/game-profile-manager.rst

```diff
@@ -0,0 +1,142 @@
+
+Retrieving by UUID
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: java
+
+    import java.util.UUID;
+
+    CompletableFuture<GameProfile> futureGameProfile =
+        profileManager.get(UUID.fromString("069a79f4-44e9-4726-a5be-fca90e38aaf5"));
+
+.. tip::
+    You can also retrieve many ``GameProfile``\ s at once using ``GameProfileManager#getAllById`` or
+    ``GameProfileManager#getAllByName``, passing in an ``Iterable<UUID>`` or ``Iterable<String>``, respectively. Both of
+    these methods return ``CompletableFuture<Collection<GameProfile>>``.
+
+Note that each of these methods return some sort of ``CompletableFuture``. This indicates that the ``GameProfile``
+(or ``Collection<GameProfile>``) may not be immediately available because of pending requests to the Mojang API. The
+`Javadocs for CompletableFuture <https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html>`_
+describe the full capabilities of the class, but we will focus on the ``get`` method for the purpose of this article.
+
+To retrieve a ``GameProfile`` from a ``CompletableFuture<GameProfile``, you can simply call the ``CompletableFuture#get``
+method.
+
+.. code-block:: java
+
+    GameProfile gameProfile = futureGameProfile.get();
+
+.. warning::
+    If the ``GameProfile`` is not immediately available (such as if the cache is not being used or does not contain the
+    ``GameProfile``), then ``get`` will wait for the future to complete. For that reason, it is not advisable to use
+    this on the main thread as it will halt the server. Alternatively, you can use the
+    ``CompletableFuture#thenAccept(Consumer<? super T>)`` method to specify a ``Consumer`` to be run upon completion.
+
+Creating GameProfiles
+=====================
+
+You can generate a new ``GameProfile`` using ``GameProfile#of(UUID, String)``. Note that the username
+does not necessarily need to correspond to the ``UUID`` of that player. Likewise, the ``UUID`` does not need to belong
+to a valid player.
+
+.. code-block:: java
+
+    GameProfile gameProfile = GameProfile.of(
+            UUID.fromString("00000000-0000-0000-0000-000000000000"),
+            "Herobrine");
+
+.. note::
+    It is not mandatory to specify the name of the ``GameProfile`` (``null`` is a valid argument).
+
+Filling GameProfiles
+====================
+
+Filling a ``GameProfile`` completes the profile by fetching information like the player's skin from the Mojang API.
+Note that if faked data like username is associated with a certain UUID, it will be replaced by the actual data from
+the Mojang API.
+
+.. code-block:: java
+
+    GameProfile filledProfile = profileManager.fill(gameProfile).get();
+
+Working with ProfileProperties
+==============================
+
+``GameProfile``\ s can be used to store arbitrary data about a player using ``ProfileProperty``\ s. However, this cannot
+not be used as a permanent data store, as the data is not persisted across server restarts. We can retrieve the properties
```

Comment:
The grammar is a bit weird here. "as the data does not persist between server restarts" sounds better to me.